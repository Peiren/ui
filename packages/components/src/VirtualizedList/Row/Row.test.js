import React from 'react';
import { shallow } from 'enzyme';

import RowRenderer from './Row.component';

describe('RowRenderer', () => {
	it('should show a skeleton during the scroll', () => {
		const Row = new RowRenderer({
			selectionToggle: null,
		});

		// when
		const wrapper = shallow(
			<Row
				className={'my-class-names'}
				index={1}
				key={18}
				parent={{}}
				style={{ background: 'red' }}
				isScrolling
			/>
		);

		// then
		expect(wrapper.node).toMatchSnapshot();
	});

	it('should show a RowSelectionRenderer if the row is selected', () => {
		// given
		const Row = new RowRenderer({
			selectionToggle: () => {},
			isSelected: () => {},
		});

		// when
		const wrapper = shallow(
			<Row
				className={'my-class-names'}
				index={1}
				key={18}
				parent={{}}
				style={{ background: 'red' }}
			/>
		);

		// then
		expect(wrapper.node).toMatchSnapshot();
	});

	it('should show a DefaultTableRowRenderer by default', () => {
		const Row = new RowRenderer({
			selectionToggle: null,
		});

		// when
		const wrapper = shallow(
			<Row
				className={'my-class-names'}
				index={1}
				key={18}
				parent={{}}
				style={{ background: 'red' }}
			/>
		);

		// then
		expect(wrapper.node).toMatchSnapshot();
	});
});
