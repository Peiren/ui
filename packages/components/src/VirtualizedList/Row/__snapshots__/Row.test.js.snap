// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowRenderer should show a DefaultTableRowRenderer by default 1`] = `
<defaultRowRenderer
  className="my-class-names"
  index={1}
  parent={Object {}}
  style={
    Object {
      "background": "red",
    }
  }
/>
`;

exports[`RowRenderer should show a RowSelectionRenderer if the row is selected 1`] = `
<RowSelection(undefined)
  className="my-class-names"
  index={1}
  parent={Object {}}
  style={
    Object {
      "background": "red",
    }
  }
/>
`;

exports[`RowRenderer should show a skeleton during the scroll 1`] = `
<div
  className="my-class-names"
  index={1}
  isScrolling={true}
  parent={Object {}}
  style={
    Object {
      "background": "red",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "width": "42%",
      }
    }
  />
</div>
`;
